src/ST/Client/FixtureController.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Baseline;
+using Oakton;
+using StoryTeller.Files;
+using StoryTeller.Model;
+using StoryTeller.Model.Persistence.DSL;
+
+namespace ST.Client
+{
+    public interface IFixtureFileObserver : IDisposable
+    {
+        void Changed(string file);
+        void Added(string file);
+        void Deleted(string file);
+    }
+
+    public interface IFixtureFileWatcher : IDisposable
+    {
+        void WriteFiles(Action action);
+        void StartWatching(string path, IFixtureFileObserver observer);
+    }
+
+    public class FixtureFileWatcher : IFixtureFileWatcher, IChangeSetHandler
+    {
+        private FileChangeWatcher _watcher;
+        private IFixtureFileObserver _observer;
+
+        public void Handle(ChangeSet changes)
+        {
+            changes.Changed.Each(x => _observer.Changed(x.Path));
+            changes.Deleted.Each(x => _observer.Deleted(x));
+            changes.Added.Each(x => _observer.Added(x.Path));
+        }
+
+        public void Dispose()
+        {
+            _watcher?.Dispose();
+        }
+
+        public void WriteFiles(Action action)
+        {
+            _watcher.Latch(action);
+        }
+
+        public void StartWatching(string path, IFixtureFileObserver observer)
+        {
+            var fullPath = path.ToFullPath();
+            _watcher = new FileChangeWatcher(fullPath, FileSet.Deep("*.fixture"), this);
```

Comment:
Let's go ahead and just use *.md so smarter editors like VS Code can apply their markdown extensions. If we're gonna keep the fixture md files in a completely different folder, we shouldn't have any trouble w/ that.